client/cli/src/lib.rs

```diff
@@ -309,25 +309,46 @@ pub fn init_logger(
 	}
 
 	let isatty = atty::is(atty::Stream::Stderr);
-	let enable_color = isatty;
 	let timer = ChronoLocal::with_format(if simple {
 		"%Y-%m-%d %H:%M:%S".to_string()
 	} else {
 		"%Y-%m-%d %H:%M:%S%.3f".to_string()
 	});
 
-	let subscriber = FmtSubscriber::builder()
+	let subscriber_builder = FmtSubscriber::builder()
 		.with_env_filter(env_filter)
-		.with_writer(std::io::stderr)
+		.with_writer(std::io::stderr as _)
 		.event_format(logging::EventFormat {
 			timer,
-			ansi: enable_color,
+			ansi: isatty,
 			display_target: !simple,
 			display_level: !simple,
 			display_thread_name: !simple,
-		})
-		.finish().with(logging::NodeNameLayer);
+		});
+	if disable_log_reloading {
+		let subscriber = subscriber_builder
+			.finish()
+			.with(logging::NodeNameLayer);
+		initialize_tracing(subscriber, tracing_receiver, profiling_targets)
+	} else {
+		let subscriber_builder = subscriber_builder.with_filter_reloading();
```

Comment:
If you move this stuff into its own if else above. You don't need to copy the last two lines of the initialization.

Reply:
I maybe misunderstood what you said, but I don't think it's possible to move the order of the `with_filter_reloading` method.